giskard/rag/question_generators/ookb_questions.py

```diff
@@ -0,0 +1,176 @@
+import uuid
+
+from ..knowledge_base import KnowledgeBase
+from .base import GenerateFromSingleQuestionMixin, _LLMBasedQuestionGenerator
+from .prompt import QAGenerationPrompt
+
+OOKB_PROMPT = """
+You are a powerful auditor and mindful judger, your role is to generate question from a given context and 
+by adding some fake or non-existing details/facts to the context to check whether the agent model you are auditing is capable of reasoning and answering questions
+which has no direct answer in the provided context.
+
+There are your tasks, you should finish them step by step:
+0. select one detail/fact from the context
+1. add some fake details/facts which is not inclued in the whole provided context but based on the detail/facts you selected in the previous step.
+2. Isolate this added fake detail/fact into a single sentence
+3. Generate a question based on the new added isolated detail/fact, fit the question in the context and can not be answered by the information in the context.
+
+The user will provide the context and give your answer after read it thoroughly.
+You will return the isolated detail/fact and the question based exclusively on the new added isolated context.
+You must output a single JSON object with keys 'selected detail/fact', 'fake detail/fact' and 'question' , without any other wrapping text or markdown and everything is in low letter. Make sure you only return valid JSON. 
+"""
+
+OOKB_QUESTION_EXAMPLE_INPUT = """Paul Graham liked to buy a baguette every day at the local market."""
```

Comment:
Use a longer paragraph, it is unlikely that the text chunks provided will only contain 1 sentence. Then, your selected fact can remain the same but it forces the model to actually extract a fact, not simple copy/paste the context